scene/gui/text_edit.cpp

```diff
@@ -2963,6 +2963,18 @@ void TextEdit::end_complex_operation() {
 	undo_stack.back()->get().chain_backward = true;
 }
 
+bool TextEdit::has_undo() const {
+	if (undo_stack_pos == nullptr) {
+		int pending = current_op.type == TextOperation::TYPE_NONE ? 0 : 1;
```

Comment:
You shouldn't need to check the `current_op.type`, as we only use the [stack size](https://github.com/godotengine/godot/blob/363d14a69159221bf3f4864800577d14349a4c57/scene/gui/text_edit.cpp#L2986) for undo.

Reply:
`LineEdit` does not need this. But for `TextEdit`, there is a `_push_current_op()` [before](https://github.com/godotengine/godot/blob/363d14a69159221bf3f4864800577d14349a4c57/scene/gui/text_edit.cpp#L2983) checking the stack size.

For example, if `current_op.type` is not checked, it'll say not undoable after typing the first character in `TextEdit`.

Reply:
Ah, missed that, in that looks good.